### ISSUE TITLE
train.py or test.py script not working with assert error with multiple GPUs

### ISSUE THREAD

**Issue by charleswg** [YanglanOu/patcher#2]
When there're multiple GPUs, the predictions part in train.py/test.py generate half of the data (for 2 GPUs) and there's an error in size mismatch between ground truth labels and predicted labels (half the size).

**Comment by YanglanOu** [comment]
Under pytorch lightning framework, we use Distributed Data Parallel (DDP) for distributed training. As it’s described in docs: 

Each GPU across each node gets its own process.
Each GPU gets visibility into a subset of the overall dataset. It will only ever see that subset.

That’s why the number of samples (both input images and labels) will be divided by num of GPUs in train.py/test.py. Also, pytorch lightning recommends to test on single device. If you still have a mismatch error, could you please provide the error message. Thanks.
